First pass at updating en/documentation/index.md (#3427)
ruby/www.ruby-lang.org: en/documentation/index.md

### Getting Started

[Try Ruby][1]
: You can try Ruby right in your browser.

[Learn to Program][8]
: A wonderful little tutorial by Chris Pine for programming newbies. If
  you don’t know how to program, start here.

[Ruby in Twenty Minutes][rubyin20]
: A small Ruby tutorial that should take no more than 20 minutes to
  complete.

[The Odin Project][odin]
: An open source full stack curriculum

[excercism][exercism]
: 120 exercises with automatic analysis and personal mentoring.

[Codecademy][codecademy]

> : Online code bootcamp with a variety of topics. Freeish.

Reviewer: ```suggestion
: Online code bootcamp with a variety of topics.
```

### Manuals / Books

#### Beginner

[Programming Ruby 3.3][pickaxe]
: The seminal work on Ruby in English. Recently updated to ruby 3.3.

[The Well-Grounded Rubyist][grounded]
: A tutorial that begins with your first Ruby program and takes you all the way to sophisticated topics like reflection, threading, and recursion.

> #### Adept

Reviewer 2: As a non-native English speaker, I think `Adept` is a relatively rare choice of word. Maybe `Advanced` will be more approachable?

Reviewer 3: Maybe `Experienced` would be better? Advanced and Expert are closer in my mind than Adept/Experienced and Expert are.

Author: Intermediate?

Reviewer 2: Intermediate sounds good to me 👍

[RubyAPI.org][rubyapi-org]
: Easily find and browse Ruby classes, modules, and methods.

[ruby-doc.org][39]
: online API documentation

[DevDocs.io][40]
: online API documentation

[Ruby QuickRef][42]

> : The ruby quick reference

Reviewer 4: ```suggestion
: The Ruby quick reference
```

Reviewer 5: I would add that this is for both the language and the standard library. Many references for other languages cover just one or the other.

[rubyreferences][43]

> : A full language reference + detailed language changelog. Fantastic.

Reviewer: ```suggestion
: A full language reference + detailed language changelog
```

>     : A full language reference + detailed language changelog. Fantastic.

Reviewer 2: Extra spacing:

```suggestion
: A full language reference + detailed language changelog. Fantastic.
```

While I agree it's a fantastic work, it feels a bit odd to add `Fantastic.` in its description?

### Style Guides

[rubystyle.guide][44]
: Rubocop's Ruby style guide

[RuboCop][45]
: Automated enforcement of their style guide.

[Shopify][46]
: Shopify's Ruby style guide

[GitLab][47]
: Gitlab's Ruby style guide

[Airbnb][48]
: AirBNB's Ruby style guide

> [w3resource][49]

Reviewer 6: Thank you for your contribution and for adding helpful links 🙌🏻  I noticed this one has significant number of ads (screenshot below). While the content looks valuable, I wonder if we should consider whether this is a good addition to be on the documentation page on the Ruby official website 🤔 

![CleanShot 2024-11-25 at 21 51 56@2x](https://github.com/user-attachments/assets/ea19e436-a80b-482d-9316-b9bf3ce69ccc)

environment with advanced features (e.g. code completion, refactoring,
testing support).

Here is a list of popular tools used by Rubyists, broken up by
learning curve:

* Easy
  * [Sublime Text][37]
  * [Visual Studio Code][vscode]
  * [Zed][zed]
* Intermediate
  * [RubyMine][27] (paid)

> * Expert

Reviewer 2: Nit: Maybe "difficult" can fit the tooling context better than "expert"?

```suggestion
* Difficult
```

  * [Emacs][20] with [Ruby mode][21] or [Enhanced Ruby mode][enh-ruby-mode]
  * [Vim][25] with [vim-ruby][26] plugin
  * [NeoVim][neovim]

All of these editors support the Language Server Protocol (LSP),
either by default or through their LSP plugins. Shopify's
[ruby-lsp][ruby-lsp] is one of the most popular language servers for
Ruby and [supports all of the above editors][shopify_lsp].

### Further Reading

[Official FAQ](/en/documentation/faq/)
: The official frequently asked questions.

> [Ruby-doc.org][34] maintains a comprehensive list of English

Reviewer 2: I think this and link 34 can be removed because it's been listed in the `Reference Documentation` section and it's not really a "further reading" anyway.

documentation sources.

### Older Reading / Resources

These links were more prominent but haven't been updated in ages.

[Ruby Koans][2]
: The Koans walk you along the path to enlightenment in order to learn
  Ruby. The goal is to learn the Ruby language, syntax, structure, and
  some common functions and libraries. We also teach you culture.

[Ruby Essentials][7]
: A free on-line book designed to provide a concise
  and easy to follow guide to learning Ruby.

[Why’s (Poignant) Guide to Ruby][5]
: An unconventional but interesting book that will teach you Ruby
  through stories, wit, and comics. Originally created by *why the lucky
  stiff*, this guide remains a classic for Ruby learners.

[Learn Ruby the Hard Way][38]
: A very good set of exercises with explanations that guide you from
  the absolute basics of Ruby all the way to OOP and web development.

> [Programming Ruby][9]
> : The seminal work on Ruby in English, this first edition of the
>   [Pragmatic Programmers’ book][10] is available for free online.

Reviewer 4: ```suggestion
[Programming Ruby][9]
: The first edition of this seminal work on Ruby in English is available for free online.
```
The current edition is already linked to in the Books section.

Here is a list of popular tools used by Rubyists:

* Easy
  * [Sublime Text][37]
  * [Visual Studio Code][vscode]
  * [Zed][zed]
* Adept
  * [RubyMine][27] (paid)
* Expert
  * [Emacs][20] with [Ruby mode][21] or [Enhanced Ruby mode][enh-ruby-mode]
  * [Vim][25] with [vim-ruby][26] plugin

>   * [NeoVim][neovim]

Reviewer 2: I'm not sure if we need or should pair editors with experience levels. For example, I started writing Ruby with Vim, then switched to NeoVim for like 6 years, and switched to VS Code like 2 years ago.

Reviewer 3: I think the idea is from an approachability standpoint instead of an end user skill level standpoint.

Author: yeah, I meant this more about learning curve of the editor

> All of these editors support the Language Server Protocol (LSP). You
> can use Shopify's [ruby-lsp][ruby-lsp] plugin for ruby support.

Reviewer 2: ```suggestion
All of these editors support the Language Server Protocol (LSP), either by default or through their LSP plugins. 
Shopify's [ruby-lsp][ruby-lsp] is one of the most popular language servers for Ruby and [supports all of the above editors](https://shopify.github.io/ruby-lsp/editors.html)
```
